### PR TITLE
Integrate admin-ui java logic into http module

### PR DESCRIPTION
**Pull request by mfussenegger**
So far we've had 2 RestFilter implementations. 1 in `core` and 1 in
`admin-ui`.
In ES52 the `RestFilter` functionality has been removed. There is a
replacement, but only a single plugin can use that functionality.

This commit moves the "serve JSON on /" functionality from `core` to
`http`. In addition, it moves all logic from `admin-ui` into `http`.
This way we get rid of a bit duplicated guice-injection magic and have
only a single `RestFilter` implementation.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -460,13 +464,10 @@ task downloadPlugins(
             into 'plugins/crate-azure-discovery'
         }
 
-        // copy crate-admin plugin
         copy {
-            from(project(':admin-ui').buildDir.path + '/crate-admin')
-            from(project(':admin-ui').buildDir.path + '/libs')
-            into 'plugins/crate-admin'
+            from("${buildDir}/crate-admin/")
+            into 'adminui'
```
Comment (mfussenegger) [inline app/build.gradle]: This is a breaking packaging change. I'm not sure if we really want to have a `adminui` folder in the "crate root directory". But it cannot be in the plugins folder any longer as it's missing a plugin description.

Any input here?

Comment (dobe) [inline app/build.gradle]: can we move it below lib somehow?

Comment (dobe) [inline app/build.gradle]: this is just about the resources for _site, right? so probably it would make sense to use the normal resource mechanism?

Comment (mfussenegger) [inline app/build.gradle]: Moved it to `lib/site` as discussed

**Pull request merged**